konrad/atmosphere.py

```diff
@@ -317,13 +317,39 @@ def get_cold_point_index(self):
 
         return np.argmin(T[plev > self.pmin])
 
-    def get_cold_point_plev(self):
+    def get_cold_point_plev(self, interpolate=False):
         """Return the cold point pressure.
 
+        Paramteres:
+            interpolate (bool): If `False` return the pressure grid value of
+                the actual coldest point. If `True` perform a quadratic fit
+                to retrieve a smoother estimate of the cold point pressure.
+
         Returns:
             float: Pressure at the cold point [Pa].
         """
-        return self['plev'][self.get_cold_point_index()]
+        if interpolate:
+            plog = np.log(self["plev"])
+            idx = self.get_cold_point_index()
+
+            # Select a pressure region around the single coldest level.
+            mask = np.logical_and(
+                plog > plog[idx] - 0.25,
+                plog <= plog[idx] + 0.25,
+            )
+
+            # Perform a quadratic fit to the T profile around the cold point.
+            popt = np.polyfit(
+                plog[mask],
+                self["T"][-1, mask],
+                deg=2,
+            )
+
+            # Return the analytical minimum of the T profile fit.
+            return np.exp(-popt[1] / (2 * (popt[0])))
```

Comment:
Okay, I made it to the same equation as you, but comments wise it could use a bit more explanation for what you are doing!